tests/target/configs/single_line_let_else_max_width/100.rs

```diff
@@ -18,4 +18,42 @@ fn main() {
     };
 
     let Some(d) = some_very_very_very_very_long_name else { return };
+
+    let Expr::Slice(ast::ExprSlice {
+        lower,
+        upper,
+        step,
+        range: _,
+    }) = slice.as_ref() else {
+        return;
+    };
```

Comment:
Glad we added this test, as I'm unsure whether this is desired behavior.

It would appear that the prescriptions in https://doc.rust-lang.org/nightly/style-guide/statements.html#else-blocks-let-else-statements do not explicitly cover the scenario where the statement is multilined prior to the the `else` block with an initializer expression that is single line.

I feel like one could read this:

> Otherwise, the else keyword and opening brace should be placed on the next line after the end of the initializer expression, and the else keyword should have the same indentation level as the let keyword.

As implying that if any of the prior scenarios aren't true, then break before the `else` keyword, but that is rather implicit. Thoughts? :thinking:

Reply:
posted in zulip for style team as well https://rust-lang.zulipchat.com/#narrow/stream/346005-t-style/topic/let-else.20style.20guide.20inconsistency.3F/near/370115141

Reply:
I'll investigate this case and look into what's goin on. Now that you point it out I think I would have expected it to be formatted like:

```rust
    let Expr::Slice(ast::ExprSlice {
        lower,
        upper,
        step,
        range: _,
    }) = slice.as_ref()
    else {
        return;
    };
```

Reply:
The changes to produce the above formatting are pretty minimal. Let me know if I should go ahead and amend this PR once you hear back from t-style.